### PR TITLE
Add example to `optuna.study.Study.user_attrs` 

### PR DESCRIPTION
**Pull request by norihitoishida**
## Motivation
This PR is related to #1334

## Description of the changes
Update docstring of `user_attrs` , referred to [Tutorial](https://optuna.readthedocs.io/en/stable/tutorial/attributes.html).

### CODE REVIEW

File: optuna/study.py
Comment (toshihikoyanase) [inline optuna/study.py:187]: I think these values are taken from [this tutorial](https://optuna.readthedocs.io/en/stable/tutorial/attributes.html), It may be difficult for readers to imagine the actual usage. I guess they describe the objective function, but this example doesn't have objective functions. So, for example, how about adding a simple objective function and explaining it with the `user_attrs`?

```python
def objective(trial):
    x = trial.suggest_float("x", 0, 1)
    y = trial.suggest_float("y", 0, 1)
    return x ** 2 + y ** 2

...

study.set_user_attr("objective function", "quadratic fuction")
study.set_user_attr("dimensions", 2)
study.set_user_attr("contributors", ["Akiba", "Sano"])
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. It added a comment on the content of `user_attrs`.

File: optuna/study.py
Comment (toshihikoyanase) [reply optuna/study.py:187]: Ah, I should review #1744 first. I'll copy this comment and ask the other reviewer about it there.

File: optuna/study.py
Comment (HideakiImamura) [inline optuna/study.py:191]: A typo: 
```suggestion
                study.set_user_attr("objective function", "quadratic function")
```

File: optuna/study.py
Comment (HideakiImamura) [inline optuna/study.py:195]: How about checking the contents of the `study.user_attrs`?
```suggestion
                assert study.user_attrs == {
                    "objective function": "quadratic function",
                    "dimensions": 2,
                    "contributors": ["Akiba", "Sano"]
                }
```

**Review by HideakiImamura — CHANGES_REQUESTED** [review]
Thanks for the PR! I have some minor comments.

File: optuna/study.py
Comment (norihitoishida) [reply optuna/study.py:195]: Nice idea!

**Comment by norihitoishida** [comment]
Please see the discussion at #1744.

**Review by HideakiImamura — APPROVED** [review]
Thanks for the update. LGTM!

File: optuna/study.py
Comment (HideakiImamura) [inline optuna/study.py:178]: I commented in https://github.com/optuna/optuna/pull/1744#discussion_r482058080, how about using `.. seealso::`?
```suggestion
        .. seealso
        
            See :func:`~optuna.study.Study.set_user_attr` for related method.
```

**Review by HideakiImamura — CHANGES_REQUESTED** [review]
Hi @norihitoishida! Sorry for the comment after my approval. I have a minor comment. Could you take a look?

File: optuna/study.py
Comment (norihitoishida) [reply optuna/study.py]: @HideakiImamura  Thanks for nice idea!

**Review by HideakiImamura — APPROVED** [review]
LGTM, again!

**Review by toshihikoyanase — APPROVED** [review]
LGTM! Thank you as always.

**Pull request merged**
